thirdparty/etcpak/patches/0001-Make-mingw-deal-with-intrinsics-better-and-narrowing.patch

```diff
@@ -0,0 +1,199 @@
++        static_cast<uint32_t>(clampu8( r1 + table59T58H[codeword] ) | ( clampu8( g1 + table59T58H[codeword] ) << 8 ) | ( clampu8( b1 + table59T58H[codeword] ) << 16 )),
++        static_cast<uint32_t>(clampu8( r1 - table59T58H[codeword] ) | ( clampu8( g1 - table59T58H[codeword] ) << 8 ) | ( clampu8( b1 - table59T58H[codeword] ) << 16 ))
+     };
+ 
+     for( uint8_t j = 0; j < 4; j++ )
+diff --git a/thirdparty/etcpak/ProcessRGB.cpp b/thirdparty/etcpak/ProcessRGB.cpp
+index 220d5c55e26..3d3ba7e7f78 100644
+--- a/thirdparty/etcpak/ProcessRGB.cpp
++++ b/thirdparty/etcpak/ProcessRGB.cpp
+@@ -12,8 +12,8 @@
+ #include "ProcessRGB.hpp"
+ #include "Tables.hpp"
+ #include "Vector.hpp"
+-#if defined __SSE4_1__ || defined __AVX2__ || defined _MSC_VER
+-#  ifdef _MSC_VER
++#if defined __SSE4_1__ || defined __AVX2__
++#  ifdef _MSC_VER && !defined(__MINGW32__)
+ #    include <intrin.h>
+ #    include <Windows.h>
+ #    define _bswap(x) _byteswap_ulong(x)
+@@ -21,17 +21,27 @@
+ #  else
+ #    include <x86intrin.h>
+ #  endif
+-#else
+-#  ifndef _MSC_VER
+-#    include <byteswap.h>
+-#    define _bswap(x) bswap_32(x)
+-#    define _bswap64(x) bswap_64(x)
+-#  endif
+ #endif
+ 
+-#ifndef _bswap
+-#  define _bswap(x) __builtin_bswap32(x)
+-#  define _bswap64(x) __builtin_bswap64(x)
++#if !defined(_bswap) && !defined(_bswap64)
++// Copyright 2020 github user jtbr, Released under MIT license
++// https://gist.github.com/jtbr/7a43e6281e6cca353b33ee501421860c
++static inline uint32_t _bswap(uint32_t x) {
++    return ((( x & 0xff000000u ) >> 24 ) |
++            (( x & 0x00ff0000u ) >> 8  ) |
++            (( x & 0x0000ff00u ) << 8  ) |
++            (( x & 0x000000ffu ) << 24 ));
++}
++static inline uint64_t _bswap64(uint64_t x) {
++    return ((( x & 0xff00000000000000ull ) >> 56 ) |
++            (( x & 0x00ff000000000000ull ) >> 40 ) |
++            (( x & 0x0000ff0000000000ull ) >> 24 ) |
++            (( x & 0x000000ff00000000ull ) >> 8  ) |
++            (( x & 0x00000000ff000000ull ) << 8  ) |
++            (( x & 0x0000000000ff0000ull ) << 24 ) |
++            (( x & 0x000000000000ff00ull ) << 40 ) |
++            (( x & 0x00000000000000ffull ) << 56 ));
++}
+ #endif
+ 
+ namespace
+diff --git a/thirdparty/etcpak/System.cpp b/thirdparty/etcpak/System.cpp
+index 1383d0ecd0e..eb4dbc2f97d 100644
+--- a/thirdparty/etcpak/System.cpp
++++ b/thirdparty/etcpak/System.cpp
+@@ -35,7 +35,7 @@ unsigned int System::CPUCores()
+ 
+ void System::SetThreadName( std::thread& thread, const char* name )
+ {
+-#ifdef _WIN32
++#if defined(_WIN32) && !defined(__MINGW32__)
+     const DWORD MS_VC_EXCEPTION=0x406D1388;
+ 
+ #  pragma pack( push, 8 )
+@@ -62,7 +62,7 @@ void System::SetThreadName( std::thread& thread, const char* name )
+     __except(EXCEPTION_EXECUTE_HANDLER)
+     {
+     }
+-#elif !defined(__APPLE__)
++#elif !defined(__APPLE__) && !defined(_WIN32) && !defined(__MINGW32__)
```

Comment:
This is not blocking for this PR, but I think *BSD systems won't be able to build etcpak as is due to this. That's something we'll have to fix before the 4.0 release as we support those platforms.

See this code, it's pretty tricky to have cross-platform:
https://github.com/godotengine/godot/blob/8b6e3d63750db5eab94b5d4e5dc182a678c1e455/drivers/unix/thread_posix.cpp#L42-L70

Reply:
Should be fixed with #47852 (doing like upstream did for Apple - no thread names on *BSD).